tensorflow_addons/image/color_ops.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Color Ops"""
+
+import tensorflow as tf
+
+from tensorflow_addons.utils.types import TensorLike
+from tensorflow_addons.image.utils import to_4D_image, from_4D_image
+
+from typing import Optional
+
+
+def equalize_batch(image: TensorLike, data_format: str = "channels_last") -> tf.Tensor:
+    """Implements Equalize function from PIL using TF ops."""
+
+    def scale_channel(image, channel=None):
+        """Scale the data in the channel to implement equalize."""
+        image_dtype = image.dtype
+        image = tf.cast(image[:, :, channel], tf.int32)
```

Comment:
`image[:, :, channel]` assumes that it's `"channels_last"`.

Reply:
Missed that. Adding test-case for `"channel_first"`.